src/main/java/gomedic/storage/JsonAdaptedDoctor.java

```diff
@@ -0,0 +1,101 @@
+package gomedic.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import gomedic.commons.exceptions.IllegalValueException;
+import gomedic.model.commonfield.Name;
+import gomedic.model.commonfield.Phone;
+import gomedic.model.person.AbstractPerson;
+import gomedic.model.person.doctor.Department;
+import gomedic.model.person.doctor.Doctor;
+import gomedic.model.person.doctor.DoctorId;
+
+/**
+ * Jackson-friendly version of {@link Doctor}.
+ */
+public class JsonAdaptedDoctor {
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Doctor's %s field is missing!";
+
+    private final String name;
+    private final String phone;
+    private final String id;
+    private final String department;
+
+    /**
+     * Constructs a {@code JsonAdaptedDoctor} with the given details.
+     */
+    @JsonCreator
+    public JsonAdaptedDoctor(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
+                             @JsonProperty("id") String id, @JsonProperty("department") String department) {
+        this.name = name;
+        this.phone = phone;
+        this.id = id;
+        this.department = department;
+    }
+
+    /**
+     * Converts a given {@code Doctor} into this class for Jackson use.
+     */
+    public JsonAdaptedDoctor(Doctor source) {
+        name = source.getName().fullName;
+        phone = source.getPhone().value;
+        id = source.getId().toString();
+        department = source.getDepartment().departmentName;
+
+    }
+
+    /**
+     * Converts a given {@code Doctor} into this class for Jackson use.
+     */
+    public JsonAdaptedDoctor(AbstractPerson source) {
```

Comment:
Hem, sorry why do we need two constructors, one for abstract person and the other for doctor ?

Reply:
Oops mb i accidentally left in the constructor that takes in a Doctor. 

Also, regarding this part, i made some comments under review point, do u mind commenting on it? `JsonAdaptedDoctor` is receiving the `Doctor` objects from a `UniqueAbstractPersonList` which can only return them as compile type `AbstractPerson`, so i need to cast them to `Doctor`

Reply:
Actually I think you can make the unique person list generic such as <? extends AbstractPerson> and this way you can create a UniquePersonList<Doctor>() and save the doctor object itself, I think this is much better that just passing the person